kong/llm/init.lua

```diff
@@ -327,16 +328,22 @@ function _M:ai_introspect_body(request, system_prompt, http_opts, response_regex
   return new_request_body
 end
 
-function _M:parse_json_instructions(body_string)
-  local instructions, err = cjson.decode(body_string)
-  if err then
-    return nil, nil, nil, err
+function _M:parse_json_instructions(in_body)
+  local err
+  if type(in_body) == "string" then
+    in_body, err = cjson.decode(in_body)
+    if err then
+      return nil, nil, nil, err
+    end
+
+  elseif type(in_body) ~= "table" then
+    return nil, nil, nil, "input not table or string"
```

Comment:
I think this needs to be moved into its own conditional construct so that it covers the already-decoded case and the decode case:


```suggestion
  end

  if type(in_body) ~= "table" then
    return nil, nil, nil, "input not table or string"
  end
```